src/mpid/ch4/netmod/ucx/ucx_progress.c

```diff
@@ -78,44 +76,31 @@ static int am_handler(void *msg, size_t msg_sz)
     if (target_cmpl_cb) {
         target_cmpl_cb(rreq);
     }
-
-  fn_exit:
-    return mpi_errno;
-}
-
-static void handle_am_recv(void *request, ucs_status_t status, ucp_tag_recv_info_t * info)
-{
-    return;
 }
 
 int MPIDI_UCX_progress(int vci, int blocking)
 {
     int mpi_errno = MPI_SUCCESS;
     ucp_tag_recv_info_t info;
     MPIDI_UCX_ucp_request_t *ucp_request;
-    void *am_buf;
     ucp_tag_message_h message_handle;
+
     /* check for active messages */
-    message_handle =
-        ucp_tag_probe_nb(MPIDI_UCX_global.worker, MPIDI_UCX_AM_TAG, MPIDI_UCX_AM_TAG, 1, &info);
-    while (message_handle) {
+    while ((message_handle =
+            ucp_tag_probe_nb(MPIDI_UCX_global.worker, MPIDI_UCX_AM_TAG, MPIDI_UCX_AM_TAG, 1,
+                             &info))) {
         am_buf = MPL_malloc(info.length, MPL_MEM_BUFFER);
         ucp_request = (MPIDI_UCX_ucp_request_t *) ucp_tag_msg_recv_nb(MPIDI_UCX_global.worker,
                                                                       am_buf,
                                                                       info.length,
                                                                       ucp_dt_make_contig(1),
-                                                                      message_handle,
-                                                                      &handle_am_recv);
+                                                                      message_handle, &am_handler);
         while (!ucp_request_is_completed(ucp_request)) {
             ucp_worker_progress(MPIDI_UCX_global.worker);
         }
 
         ucp_request_release(ucp_request);
-        am_handler(am_buf, info.length);
```

Comment:
I guess the question is do we desire `am_handler` call to be inlined?

Reply:
Fair question. I suppose we could try and measure how the compiler does between this and the previous version. I am curious how the empty callback function actually looked at runtime. Unfortunately there is no way to supply a `NULL` callback function.

Reply:
Handling an active message receive, as an example, is roughly 375 instructions based on tracing. IMO, inlining will not provide any real benefit.

Also note that UCX is forced to call the recv callback function even if it is empty, so the added overhead is mostly the additional stack operations of the `am_handler` function.